_health-care/_js/_components/full_name.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import _ from 'lodash';
+
+class FullName extends React.Component {
+  constructor() {
+    super();
+    this.state = {hasError: false};
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('date-input-');
+  }
+
+  handleChange() {
+    const name = {
+      first: this.refs.first.value,
+      middle: this.refs.middle.value,
+      last: this.refs.last.value,
+      suffix: this.refs.suffix.value
+    }
+
+    if (!this.validate(name.first, name.last)) {
```

Comment:
I'm wondering if we need to change this paradigm of validating the entire component. This doesn't allow us to validate individual inputs within the component. This is specifically a problem for how to display the error message class (can't add the class to a particular input dependent on the state of the entire component).